### PR TITLE
cmd: order bpf policy get by DIRECTION and IDENTITY

### PR DESCRIPTION
**Pull request by aanm**
Signed-off-by: André Martins <andre@cilium.io>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5554)

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Review by joestringer — APPROVED** [review]

File: pkg/maps/policymap/policymap.go [hunk 1]
```diff
@@ -89,6 +89,19 @@ type PolicyEntryDump struct {
 	Key PolicyKey
 }
 
+// PolicyEntriesDump is a wrapper for a slice of PolicyEntryDump
+type PolicyEntriesDump []PolicyEntryDump
+
+// Less returns true if the element in index `i` is lower than the element
```
Comment (ianvernon) [inline pkg/maps/policymap/policymap.go]: This comment isn't helpful, as that is the generic definition of what `Less` is for anything which implements the Sort interface. Update it to specify that it sorts by traffic direction, and then by identity.

File: pkg/maps/policymap/policymap.go [hunk 2]
```diff
@@ -89,6 +89,19 @@ type PolicyEntryDump struct {
...
+// in index `j`
+func (p PolicyEntriesDump) Less(i, j int) bool {
```
Comment (ianvernon) [inline pkg/maps/policymap/policymap.go]: Add a unit test for this to ensure that sorting behavior isn't broken in the future.

**Review by ianvernon — DISMISSED** [review]

**Comment by ianvernon** [comment]
Thanks for adding this; it's something I've wanted to do for a long time. It will help out with debugging policy-related issues immensely!

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
possible flake: https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/6494/testReport/junit/k8s-1/8/K8sServicesTest_Bookinfo_Demo_Tests_bookinfo_demo/

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Review by ianvernon — APPROVED** [review]

**Review by tgraf — APPROVED** [review]

**Pull request merged**